tools/nnyeah/nnyeah/AssemblyComparator/TypeReworker.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using Mono.Cecil;
+using Mono.Cecil.Rocks;
+using System.Collections.Generic;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.MaciOS.Nnyeah;
+
+#nullable enable
+
+namespace Microsoft.MaciOS.AssemblyComparator {
+	public class TypeReworker {
+		ModuleDefinition module;
+		TypeDefinition nintTypeReference;
+		TypeDefinition nuintTypeReference;
+		TypeDefinition nfloatTypeReference;
+		TypeReference newNfloatTypeReference;
+		ModuleReference newNfloatModuleReference;
+		TypeReference newNHandleTypeReference;
+
+		public TypeReworker (ModuleDefinition module)
+		{
+			this.module = module;
+			nintTypeReference = GetTypeByName (module, "System.nint");
+			nuintTypeReference = GetTypeByName (module, "System.nuint");
+			nfloatTypeReference = GetTypeByName (module, "System.nfloat");
+			newNfloatModuleReference = new ModuleReference ("System.Private.CoreLib");
+			newNfloatTypeReference = new TypeReference ("System.Runtime.InteropServices",
+				"NFloat", null, newNfloatModuleReference, true);
+			newNHandleTypeReference = new TypeReference ("ObjCRuntime", "NativeHandle",
+				null, newNfloatModuleReference, true);
+		}
+
+		static TypeDefinition GetTypeByName (ModuleDefinition module, string name)
+		{
+			foreach (var ty in module.Types) {
+				if (ty.ToString () == name)
+					return ty;
+			}
+			throw new Exception (String.Format(Errors.E0010, name));
```

Comment:
Style, but I prefer the LINQ search again. 

```suggestion
			var type = module.Types.FirstOrDefault (t => t.ToString() == name);
			if (type == null) {
				throw new Exception (String.Format(Errors.E0010, name));
			}
			return type;
```